File: frontend/src/types/stats.ts
import type { HistoryItem } from "./history";

export interface DashboardStats {
  total_records: number;
  total_images: number;
  total_videos: number;
  total_detections: number;
  total_fire: number;
  total_smoke: number;
  today_count: number;
  detections_by_day: { date: string; count: number }[];
  confidence_distribution: Record<string, number>;
  class_pie: { fire: number; smoke: number };
  avg_confidence_fire: number | null;
  avg_confidence_smoke: number | null;
  recent_activity: HistoryItem[];
}

File: backend/app/schemas/stats.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class DashboardStats(BaseModel):
    total_records: int = 0
    total_images: int = 0
    total_videos: int = 0
    total_detections: int = 0
    total_fire: int = 0
    total_smoke: int = 0
    today_count: int = 0
    detections_by_day: list[dict] = []
    confidence_distribution: dict[str, int] = {}
    class_pie: dict[str, int] = {}
    avg_confidence_fire: float | None = None
    avg_confidence_smoke: float | None = None
    recent_activity: list[dict] = []
